src/peft/tuners/tuners_utils.py

```diff
@@ -902,6 +953,7 @@ def _move_adapter_to_device_of_base_layer(self, adapter_name: str, device: Optio
             if any(p.device == meta for p in adapter_layer.parameters()):
                 continue
 
+            # TODO: weight is not necessarily defined here, leading to a NameError, fix that
```

Comment:
Note: This is an existing bug and has nothing to do with the PR, just flagging it here.